app/frontend/src/store/modules/ches.js

```diff
@@ -71,38 +84,45 @@ export default {
       commit('ADD_TXID', txId);
     },
 
-    addTableData({ commit }, data) {
-      commit('ADD_TABLEDATA', data);
-    },
-    // commit,
-    populateTable({ commit, state }) {
-      try {
-        // for each existing tx in txIds array
-        state.txIds.forEach(async (tx) => {
-          // if tableData doesnt contain an object with txId = tx.txId
-          if (!state.tableData.find((o) => o.txId === tx.txId)) {
-            // for each of the messages in tx
-            tx.messages.forEach(async (message) => {
-              // get details from CHES and add to tableData
+    // populates history table each time the component is mounted
+    async populateTable({ commit, state, dispatch }) {
+
+      commit('UPDATE_TABLELOADING', true);
+
+      // for each of the transactions that exists in local storage
+      state.txIds.forEach( (tx) => {
+        // for each of the messages in tx
+        tx.messages.forEach(async (message) => {
+          // if existing tableData doesnt contain this transaction, and the message in this row of table has a status that is not 'completed' or 'cancelled'
+          if (!state.tableData.find((o) => (o.txId === tx.txId && (o.status === 'completed' || o.status === 'cancelled')))) {
+
+            try {
+              // get details for that message (status, smpt response, etc) from CHES and update tableData
+              // note: this call tyo CHES api returns an array containing a single object
               const { data } = await chesService.getStatusByMessageId(
                 message.msgId
               );
```

Comment:
Can probably collapse this into one line - don't need to expand out the single argument in here.